### PR TITLE
[ITSASU-1757] Add sort functionality to Glossary page

### PR DESCRIPTION
**Pull request by aug24**

### CODE REVIEW

File: app/uk/gov/hmrc/incometaxsoftwarechoicesfrontend/controllers/GlossaryController.scala [hunk 1]
```diff
@@ -39,31 +40,31 @@ class GlossaryController @Inject()(mcc: MessagesControllerComponents,
 
   def show: Action[AnyContent] = Action { implicit request =>
     implicit val requestLanguage: Lang = request.lang(messagesApi)
-    val glossaryList: List[(String, List[(String, String)])] = glossaryService.getGlossaryList
+    val glossaryContent: GlossaryContent = glossaryService.getGlossaryContent()
```
Comment (aug24) [inline app/uk/gov/hmrc/incometaxsoftwarechoicesfrontend/controllers/GlossaryController.scala:43]: Introduced a new type,`GlossaryContent`, for simplicity, and renamed parameters to match everywhere.

File: app/uk/gov/hmrc/incometaxsoftwarechoicesfrontend/services/GlossaryService.scala [hunk 2]
```diff
@@ -60,34 +59,33 @@ class GlossaryService @Inject()(messagesApi: MessagesApi, langs: Langs, language
       .toSet
   }
 
-  private def getMessagesPairs(lang: Lang): List[(String, List[(String, String)])] = {
+  private def getAllMessagesPairsUnsorted(lang: Lang): GlossaryContent = {
```
Comment (aug24) [inline app/uk/gov/hmrc/incometaxsoftwarechoicesfrontend/services/GlossaryService.scala:62]: Name changed to be more explicit.  No point in sorting at this stage, as we will always sort at fetch time.

File: app/uk/gov/hmrc/incometaxsoftwarechoicesfrontend/services/GlossaryService.scala [hunk 3]
```diff
@@ -96,6 +94,16 @@ object GlossaryService {
 
   }
 
+  private[services] def sortSearchResults(maybeSortOrder: Option[String])(glossaryContent: GlossaryContent): GlossaryContent = maybeSortOrder match {
```
Comment (aug24) [inline app/uk/gov/hmrc/incometaxsoftwarechoicesfrontend/services/GlossaryService.scala:97]: The actual sort functions!

File: app/uk/gov/hmrc/incometaxsoftwarechoicesfrontend/views/GlossaryPage.scala.html [hunk 4]
```diff
@@ -70,26 +70,43 @@ <h1 class="inverse-header govuk-heading-xl">@messages("glossary.title")</h1>
     </div>
 }
 
-@inputForm = {
-  @form(action = postAction) {
+@errors = {
```
Comment (aug24) [inline app/uk/gov/hmrc/incometaxsoftwarechoicesfrontend/views/GlossaryPage.scala.html:73]: Little bit of page restructuring to make it easier to navigate.

**Review by srikanthsunkara79 — APPROVED** [review]

**Pull request merged**